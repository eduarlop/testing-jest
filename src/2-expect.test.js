const object = {
  id: 1234,
  name: "foo bar",
  skills: ["code", "teach", "learn"],
  url: "https://github.com/foobar",
  preferences: {
    theme: "dark"
  },
  active: true,
  isSubscribed: false
};

describe("Expect examples", () => {
  it("checks against expected values", () => {
    // Strict equality check
    expect(object.id).toBe(1234);

    // Compares keys recursively
    expect(object.preferences).toEqual({ theme: "dark" });

    // Checks if a value is defined
    expect(object.name).toBeDefined();

    // Checks if a value is thruty; e.g. true, 1, 'abc'
    expect(object.active).toBeTruthy();

    // Checks if a value is falsy; e.g. false, 0, null
    expect(object.isSubscribed).toBeFalsy();

    // Checks if value is of a given instance
    expect(object.skills).toBeInstanceOf(Array);

    // Checks if value is in an array
    // Also can check if a string is a substring of the other
    expect(object.skills).toContain("teach");

    // Checks if value is in an array, comparing key and values
    expect(object.name).toContain("bar");

    // Useful if the output is not deterministic on runtime
    // so you don't know what to check against
    // (however your test output SHOULD be deterministic)
    expect(object).toMatchSnapshot();
  });

  it("checks on data structures", () => {
    expect(object).toEqual({
      id: expect.any(Number),
      name: expect.stringContaining("foo"),
      skills: expect.arrayContaining([expect.any(String)]),
      url: expect.stringMatching(/^https?:\/\//i),
      preferences: expect.objectContaining({
        theme: expect.stringMatching(/^light|dark$/)
      }),
      active: expect.any(Boolean),
      isSubscribed: expect.any(Boolean)
    });
  });

  it("partially checks data structures", () => {
    expect(object).toEqual(
      expect.objectContaining({
        id: expect.any(Number),
        name: expect.stringContaining("foo"),
        active: expect.any(Boolean)
      })
    );
  });
});
