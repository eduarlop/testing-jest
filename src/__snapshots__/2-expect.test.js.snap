// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect examples checks against expected values 1`] = `
Object {
  "active": true,
  "id": 1234,
  "isSubscribed": false,
  "name": "foo bar",
  "preferences": Object {
    "theme": "dark",
  },
  "skills": Array [
    "code",
    "teach",
    "learn",
  ],
  "url": "https://github.com/foobar",
}
`;
